docs/content/concepts/assets/external-assets.mdx

```diff
@@ -0,0 +1,192 @@
+- Use the UI or GraphQL API to instigate ad hoc materializations.
+
+## Relevant APIs
+
+| Name                                             | Description                                                                             |
+| ------------------------------------------------ | --------------------------------------------------------------------------------------- |
+| <PyObject object="external_assets_from_specs" /> | Create a list of external asset definitions from a list of specs                        |
+| <PyObject object="AssetSpec" />                  | An object that represents the metadata of a particular asset                            |
+| <PyObject object="SensorResult" />               | Result object for a sensor evalation where you pass events to attach to external assets |
+| <PyObject object="DagsterInstance" />            | Object presenting the Dagster instance                                                  |
+
+## Defining External Assets
+
+### A single external asset
+
+The following code declares a single external asset and passes it to <PyObject object="Definitions"/>
+
+```python file=/concepts/assets/external_assets/single_declaration.py
+from dagster import AssetSpec, Definitions, external_asset_from_spec
+
+defs = Definitions(assets=[external_asset_from_spec(AssetSpec("my_asset"))])
+```
+
+### External assets with dependencies
+
+External assets can depend on other external assets. You do this by using the `deps` argument of <PyObject object="AssetSpec" />.
+
+```python file=/concepts/assets/external_assets/external_asset_deps.py
+from dagster import AssetSpec, Definitions, external_assets_from_specs
+
+asset_one = AssetSpec("asset_one")
+asset_two = AssetSpec("asset_two", deps=[asset_one])
+
+defs = Definitions(assets=external_assets_from_specs([asset_one, asset_two]))
+```
+
+## Keeping an external asset's metadata up to date.
+
+Since with external assets Dagster does not control scheduling or materialization, it is up to the user to keep the metadata for an external asset up-to-date. There are a number of mechanisms for doing so:
+
+- REST APIs
+- Sensors
+- Log events in ops
+
+### Using REST API
+
+Dagster OSS exposes a REST endpoint for reporting asset materializations.
+
+Here's an example `curl` command in a shell script.
+```
+curl --request POST \
+  --url https://path/to/instance/report_asset_materialization/{asset_key}\
+  --header 'Content-Type: application/json' \
+  --data '{
+	"metadata" : {
+		"source": "From curl command,
+	}
+}'
+```
+
+Here's an example of invoking this API in Python with `requests`:
+
+```python
+import requests
+
+url = f"https://path/to/instance/report_asset_materialization/{asset_key}"
+payload = { "metadata": { "source": "From python script" } }
+headers = { "Content-Type": "application/json" }
+
+response = requests.request("POST", url, json=payload, headers=headers)
+```
+
+There are also end points for reporting asset observations and asset check evaluations.
```

Comment:
Do we have these endpoints documented anywhere?

Reply:
+1

It feels weird to include this info and then not expand on it or link to supporting docs. If we don't have docs or examples for these yet (and won't before launch) I suggest we remove this bit until we do

Reply:
The docs don't exist yet @Ramshackle-Jamathon @alangenfeld one of you two should do that this week